src/CommandRunner.cpp

```diff
@@ -311,35 +311,41 @@ CommandRunner::CommandRunner(bool debug,
  *
  * @return The output of the command read from the output file.
  */
-Try<std::string> CommandRunner::run(const Command& command,
-                                    const std::string& input) {
+Try<string> CommandRunner::run(const Command& command,
+                               const std::string& input) {
   try {
-    TemporaryFile inputFile;
-    TemporaryFile outputFile;
-    inputFile.write(input);
+    TemporaryFile stdinFile;
+    TemporaryFile stdoutFile;
+    TemporaryFile stderrFile;
+    stdinFile.write(input);
 
     if (m_debug) {
       TASK_LOG(INFO, m_loggingMetadata)
           << "Calling command: \"" << command.command() << "\" ("
-          << command.timeout() << "s) " << inputFile.filepath() << " "
-          << outputFile.filepath();
+          << command.timeout() << "s) " << stdinFile.filepath() << " "
+          << stdoutFile.filepath() << " " << stderrFile.filepath();
     } else {
       TASK_LOG(INFO, m_loggingMetadata)
           << "Calling command: \"" << command.command() << "\" ("
           << command.timeout() << "s)";
     }
 
     vector<string> args;
-    args.push_back(inputFile.filepath());
-    args.push_back(outputFile.filepath());
+    args.push_back(stdinFile.filepath());
+    args.push_back(stdoutFile.filepath());
+    args.push_back(stderrFile.filepath());
 
     Try<Nothing> result = runCommandWithTimeout(
         command.command(), args, command.timeout(), m_loggingMetadata);
     if (result.isError()) {
+      string stderr = stderrFile.readAll();
+      if (!stderr.empty()) {
+        throw std::runtime_error(result.error() + " Cause: " + stderr);
```

Comment:
isn't this too long? Shouldn't we make sure we truncate stderr to a fixed number of chars?

Reply:
I will rename stdXXX by something else because they are not really log containers. input must contain the input JSON, output must contain the output JSON and err must contain an exception message (and not error logs really).
The error logs put in the sandbox are actually published by the Ruby isolator instead.

Reply:
fixed

Reply:
fixed